src/realm/unicode.cpp

```diff
@@ -358,28 +285,41 @@ util::Optional<std::string> case_map(StringData source, bool upper)
     result.resize(source.size());
 
 #if defined(_WIN32)
+    constexpr size_t tmp_buffer_size = 32;
     const char* begin = source.data();
     const char* end = begin + source.size();
     auto output = result.begin();
     while (begin != end) {
-        int n = static_cast<int>(sequence_length(*begin));
-        if (n == 0 || end - begin < n)
```

Comment:
We still need something like this check don't we? What if there are random bytes at the end of the string which appear to be a multibyte character but are just garbage such that `sequence_length()` returns 4 but there is only one character left in the string?

Reply:
Changed the logic a bit so that trailing garbage should be handled. I did not have a chance to test it on a Windows machine today. I will do that tomorrow.

Reply:
Thanks, I'd be more convinced that this works if there was a test for trailing garbage as well 😄

Reply:
Perhaps we should not specify how trailing garbage would be treated and fall back to "garbage in -> garbage out". This is anyway how the non-windows version works.

Reply:
Sure, let's match the non-windows version. Can you add a test with trailing garbage to check the consistency though?

Reply:
I don't think I can make a test that has the same outcome on both Windows and Linux. On Linux we will just copy over anything that is not an ASCII letter, but on Windows it will fail if there is something that cannot be translated. That was what I meant with my comment: We should not specify how it behaves as is behaves differently.